File: src/RealEstateCRM.Domain/Entities/Payment.cs
using RealEstateCRM.Domain.Common;
using RealEstateCRM.Domain.Enums;

namespace RealEstateCRM.Domain.Entities;

/// <summary>An online payment (e.g. a deal's down payment) collected via a payment gateway (Stripe).</summary>
public class Payment : TenantEntity
{
    public Guid DealId { get; set; }
    public decimal Amount { get; set; }
    public string Currency { get; set; } = "usd";
    public PaymentStatus Status { get; set; } = PaymentStatus.Pending;
    public string? GatewayCheckoutSessionId { get; set; }
    public string? GatewayPaymentIntentId { get; set; }
    public Guid CreatedByUserId { get; set; }
    public DateTime? PaidAt { get; set; }
}

File: client/real-estate-crm-react/src/types/payment.ts
export const PaymentStatus = {
  Pending: "Pending",
  Paid: "Paid",
  Failed: "Failed",
  Cancelled: "Cancelled",
} as const;
export type PaymentStatus = (typeof PaymentStatus)[keyof typeof PaymentStatus];

export interface Payment {
  id: string;
  dealId: string;
  amount: number;
  currency: string;
  status: PaymentStatus;
  paidAt: string | null;
  createdAt: string;
}

export interface CreateCheckoutRequest {
  amount?: number;
  currency?: string;
}

export interface CheckoutSession {
  paymentId: string;
  checkoutUrl: string;
}
